airbyte-integrations/connectors/destination-iceberg-v2/src/test-integration/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergV2DataDumper.kt

```diff
@@ -0,0 +1,142 @@
+
+        val syncId = airbyteMeta.getField("sync_id") as? Long
+        val inputChanges =
+            airbyteMeta.getField("changes") as? List<Record>
+                ?: throw IllegalStateException("Received no changes in the metadata.")
+
+        val metaChanges =
+            inputChanges.map { change ->
+                val field = change.getField("field") as String
+                val changeValue =
+                    AirbyteRecordMessageMetaChange.Change.fromValue(
+                        change.getField("change") as String
+                    )
+                val reason =
+                    AirbyteRecordMessageMetaChange.Reason.fromValue(
+                        change.getField("reason") as String
+                    )
+                DestinationRecord.Change(field, changeValue, reason)
+            }
+
+        return OutputRecord.Meta(syncId = syncId, changes = metaChanges)
+    }
+
+    override fun dumpRecords(
+        spec: ConfigurationSpecification,
+        stream: DestinationStream
+    ): List<OutputRecord> {
+        val config =
+            IcebergV2ConfigurationFactory()
+                .makeWithoutExceptionHandling(spec as IcebergV2Specification)
+        val pipeline = ParquetMapperPipelineFactory().create(stream)
+        val schema = pipeline.finalSchema as ObjectType
+        val catalog = getNessieCatalog(config)
+        val table =
+            catalog.loadTable(
+                TableIdentifier.of(stream.descriptor.namespace, stream.descriptor.name)
+            )
+
+        val outputRecords = mutableListOf<OutputRecord>()
+        IcebergGenerics.read(table).build().use { records ->
+            for (record in records) {
+                outputRecords.add(
+                    OutputRecord(
+                        rawId =
+                            UUID.fromString(
+                                record
+                                    .getField(DestinationRecord.Meta.COLUMN_NAME_AB_RAW_ID)
+                                    .toString()
+                            ),
+                        extractedAt =
+                            Instant.ofEpochMilli(
+                                record.getField(DestinationRecord.Meta.COLUMN_NAME_AB_EXTRACTED_AT)
+                                    as Long
+                            ),
+                        loadedAt = null,
+                        generationId =
+                            record.getField(DestinationRecord.Meta.COLUMN_NAME_AB_GENERATION_ID)
+                                as Long,
+                        data = getCastedData(schema, record),
+                        airbyteMeta = getMetaData(record)
+                    )
+                )
+            }
+        }
+
+        catalog.close()
+        return outputRecords
+    }
+
+    override fun dumpFile(
+        spec: ConfigurationSpecification,
+        stream: DestinationStream
+    ): List<String> {
+        TODO("Not yet implemented")
```

Comment:
```suggestion
        throw NotImplementedError("Iceberg doesn't support universal file transfer")
```
afaik we'll never implement this